tests/functional/spec/features/anchor_scrolling.spec.js

```diff
@@ -0,0 +1,71 @@
+import InsuranceAddressPage from "../../generated_pages/anchor_scrolling/insurance-address.page.js";
+import InsuranceTypePage from "../../generated_pages/anchor_scrolling/insurance-type.page.js";
+import PropertyDetailsSummaryPage from "../../generated_pages/anchor_scrolling/property-details-section-summary.page.js";
+import HouseType from "../../generated_pages/anchor_scrolling/house-type.page.js";
+import HouseholdDetailsSummaryPage from "../../generated_pages/anchor_scrolling/house-details-section-summary.page.js";
+import SubmitPage from "../../generated_pages/anchor_scrolling/submit.page.js";
+import AddressDurationPage from "../../generated_pages/anchor_scrolling/address-duration.page.js";
+
+describe("Summary Anchor Scrolling", () => {
+  describe("Given I start a Test Section Summary survey", () => {
+    beforeEach(() => {
+      browser.openQuestionnaire("test_anchor_scrolling.json");
+      $(InsuranceTypePage.both()).click();
+      $(InsuranceTypePage.submit()).click();
+    });
+
+    it("When I have provided an answer and click through to the next question, Then the Previous link url shouldn't contain any anchors or reference to return_to or return_to_answer_id", () => {
+      expect($(InsuranceAddressPage.previous()).getAttribute("href")).not.to.contain("#");
+      expect($(InsuranceAddressPage.previous()).getAttribute("href")).not.to.contain("return_to");
+      expect($(InsuranceAddressPage.previous()).getAttribute("href")).not.to.contain("return_to_answer_id");
+    });
+
+    it("When I reach the section summary page, Then the Change link url should contain return_to, return_to_answer_id query params", () => {
+      $(InsuranceAddressPage.submit()).click();
+      $(AddressDurationPage.submit()).click();
+      expect($(PropertyDetailsSummaryPage.insuranceAddressAnswer2Edit()).getAttribute("href")).to.contain(
+        "/questionnaire/insurance-address/?return_to=section-summary&return_to_answer_id=insurance-address-answer2#insurance-address-answer2"
```

Comment:
I think for these tests, the asserts could be slimmed down, to just check everything from `?`, as the URL leading to it isn't important. That should be tested already, if not should be separate which ensures the previous link is working. This to me is testing the query param + anchor.